packages/@react-aria/overlays/src/useOverlayPosition.ts

```diff
@@ -115,20 +115,27 @@ export function useOverlayPosition(props: AriaPositionProps): PositionAria {
       return;
     }
 
-    setPosition(
-      calculatePosition({
-        placement: translateRTL(placement, direction),
-        overlayNode: overlayRef.current,
-        targetNode: targetRef.current,
-        scrollNode: scrollRef.current,
-        padding: containerPadding,
-        shouldFlip,
-        boundaryElement,
-        offset,
-        crossOffset,
-        maxHeight
-      })
-    );
+    let position = calculatePosition({
+      placement: translateRTL(placement, direction),
+      overlayNode: overlayRef.current,
+      targetNode: targetRef.current,
+      scrollNode: scrollRef.current,
+      padding: containerPadding,
+      shouldFlip,
+      boundaryElement,
+      offset,
+      crossOffset,
+      maxHeight
+    });
+
+    // Modify overlay styles directly so positioning happens immediately without the need of a second render
+    // This is so we don't have to delay autoFocus scrolling or delay applying preventScroll for popovers
+    let positionPx = {maxHeight: position.maxHeight != null ? position.maxHeight + 'px' : undefined};
+    Object.keys(position.position).forEach(key => positionPx[key] = position.position[key] + 'px');
```

Comment:
heh, when I tried doing `Object.assign(overlayRef.current.style, {...position.position, maxHeight: position.maxHeight})`  it actually didn't work because the values in `position.position` are all numbers and not "px" values and thus the styles weren't actually applied at all.

Reply:
Yeah I meant like

```js
Object.keys(position.position).forEach(key => overlayRef.current.style[key] = position.position[key] + 'px');
```